src/vmm/src/guest_config/static_templates/mod.rs

```diff
@@ -1,40 +0,0 @@
-// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
-// SPDX-License-Identifier: Apache-2.0
-
-#[cfg(target_arch = "x86_64")]
-use crate::arch_gen::x86::msr_index::MSR_IA32_ARCH_CAPABILITIES;
-
-/// Hardcoded C3 template
-#[cfg(target_arch = "x86_64")]
-pub mod c3;
-/// Hardcoded T2 template
-#[cfg(target_arch = "x86_64")]
-pub mod t2;
-/// Hardcoded T2a template
-#[cfg(target_arch = "x86_64")]
-pub mod t2a;
-/// Hardcoded T2cl template
-#[cfg(target_arch = "x86_64")]
-pub mod t2cl;
-/// Hardcoded T2s template
-#[cfg(target_arch = "x86_64")]
-pub mod t2s;
-
-/// Tolerance for TSC frequency expected variation.
-/// The value of 250 parts per million is based on
-/// the QEMU approach, more details here:
-/// https://bugzilla.redhat.com/show_bug.cgi?id=1839095
-#[cfg(target_arch = "x86_64")]
-pub const TSC_KHZ_TOL: f64 = 250.0 / 1_000_000.0;
-
-#[allow(clippy::missing_docs_in_private_items)]
-#[cfg(target_arch = "x86_64")]
-static EXTRA_MSR_ENTRIES: &[u32] = &[MSR_IA32_ARCH_CAPABILITIES];
-
-/// Return a list of MSRs specific to this T2S template.
-#[inline]
-#[must_use]
-#[cfg(target_arch = "x86_64")]
-pub fn msr_entries_to_save() -> &'static [u32] {
-    EXTRA_MSR_ENTRIES
-}
```

Comment:
This code seems to be gone in commit c2e0684e901467aaaca00ce2618c08405dd92766 and be re-added in commit 900989c1ba07ce6c17d64b7905d45a9fc61006ae.  Due to this, the build on commit c2e0684e901467aaaca00ce2618c08405dd92766 fails. Could you please make no functional changes in commit c2e0684e901467aaaca00ce2618c08405dd92766 introducing and make all commits be able to build?

Reply:
I was not testing each commit for compilation and think it is a waste of time. The reason this code was removed, because the file itself was removed.

Reply:
I don't think it's waste of time. Our contribution guide include quality standards to pass tests on each commit, which means each commit is buildable.
https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards

Reply:
I guess one of the reasons for this requirement is it allows to use git bisect if we need to. Since the PR boundaries are lost when the code is merged, git bisect is not aware which commits are working and which are not.

Reply:
If we only enforce that PRs build, then the git history becomes very difficult to navigate. To benchmark or analyze any commit you need to cross-reference with closed PRs.

Reply:
Well, this does not change my opinion, but I guess I will fix this.

Reply:
> Combination of `each commit does a small change` and `making each commit in a PR to pass` tests is waste of time for a person who makes changes, because instead of making the changes he needs to think about tests.

In addition to what Nikita and Jonathan mentioned about maintainability, I think there are some benefits in review process.

In relatively big PR, it is quite difficult for reviewers (people who don't develop the code) to review only seeing diff of the PR. Instead, reviewers tend to see each commit, understand why it is required in the commit message and check how the introduced change achieve the purpose. That is why it's important to make commits small.

And especially for commits just moving some functionalities, reviews can rely on the fact that all tests pass with the commit.

Reply:
Let's just say that in my mind PR is a single change and separation into commits does not matter that much. Personally I always look at the final diffs rather then individual commits, because it gives me a better view. But anyway, I did as you asked.

Reply:
If a PR is 5+ commits, it's often almost impossible to review it altogether to a sensible quality (at least for me).
A PR is not a "merge and forget" thing. Git history is in many cases the only reliable (and available) source of information of how the product evolved and what drove particular decisions. I can remember a number of examples where we could not understand where a specific piece of code came from just because the commit message was not descriptive enough and all the authors and reviewers are long gone.